rpcserver.go

```diff
@@ -1103,3 +1108,72 @@ func (r *rpcServer) SendAsset(ctx context.Context,
 		TotalFeeSats: int64(resp.TotalFees),
 	}, nil
 }
+
+// SubscribeSendAssetEventNtfns registers a subscription to the event
+// notification stream which relates to the asset sending process.
+func (r *rpcServer) SubscribeSendAssetEventNtfns(
+	in *tarorpc.SubscribeSendAssetEventNtfnsRequest,
+	ntfnStream tarorpc.Taro_SubscribeSendAssetEventNtfnsServer) error {
+
+	eventSubscriber := chanutils.NewEventReceiver[tarofreighter.Event](
+		chanutils.DefaultQueueSize,
+	)
+	err := r.cfg.ChainPorter.RegisterSubscriber(eventSubscriber, false, false)
+	if err != nil {
+		return err
+	}
+
+	// This goroutine will read from the ChainPorter event subscription and
+	// forward to the RPC stream.
+	errorsChan := make(chan error, 1)
+	go func() {
+		for {
+			select {
+			case event := <-eventSubscriber.NewItemCreated.ChanOut():
+
+				rpcEvent, err := marshallSendAssetEvent(event)
+				if err != nil {
+					err := fmt.Errorf("failed to marshall"+
+						"ChainPorter event into RPC "+
+						"event: %w", err)
+					errorsChan <- err
+					return
+				}
+
+				err = ntfnStream.Send(rpcEvent)
+				if err != nil {
+					err := fmt.Errorf("failed to RPC "+
+						"stream send event: %w", err)
+					errorsChan <- err
+					return
+				}
+			}
```

Comment:
Same here re select on the main quit statement. We can probably also just use the context from the `in` here too.

Reply:
I've added
```
case <-r.quit:
    return
```
to the goroutine's switch statement in the latest commit. Is that the correct change?

Reply:
In case the return/output stream is closed from the client side we need to catch `case <-ntfnStream.Context().Done():` as well. I think that's what Laolu was referring to.

Reply:
Ah good point. Thank you both.